Source/Parser/PyParser.py

```diff
@@ -86,92 +81,79 @@ def getNextToken(theStream):
         matched = False
         pair = (toReturn, registeredKeywordStrings.get(toReturn))
         if pair[1] is not None:
             pair[1](toReturn, theStream)
             matched = True
-            
+
         if not matched:
             for registration in generatedRegexes:
                 if re.match(registration[0], toReturn):
                     registration[1](toReturn, theStream)
                     matched = True
                     break
-                
-        if not matched:
-            for registration in registeredRegexes:
-                if re.match(registration[0], toReturn):
-                    registration[1](toReturn, theStream)
-                    matched = True
-                    break
+
         if not matched:
             gotToken = True
-    if not toReturn=='':
+    if not toReturn == '':
         return toReturn
     return None
-                
-        
+
 
 def getNextTokenWithoutMatching(theStream):
     toReturn = ''
     gotToken = False
     while not gotToken:
-        if (not peek_char(theStream)): #I use this instead of eof
+        if not peek_char(theStream):  # I use this instead of eof
             return None
         toReturn = getNextLexeme(theStream)
         gotToken = True
-    if not toReturn=='':
+    if not toReturn == '':
         return toReturn
     return None
 
 
-
 def parseStream(theStream):
     braceDepth = 0
     for keywordItr in registeredKeywordRegexes:
-        pair = (keywordItr[0], keywordItr[1]) #keywordItr[0] is regex, keywordItr[1] is parsingfunction
+        pair = (keywordItr[0], keywordItr[1])  # keywordItr[0] is regex, keywordItr[1] is parsingfunction
         generatedRegexes.append(pair)
 
-    while(True):
+    while True:
         token = getNextToken(theStream)
-        if (token):
+        if token:
             if token == '=':
                 continue
             if token == '{':
                 braceDepth += 1
             elif token == '}':
                 braceDepth -= 1
-                if braceDepth==0:
+                if braceDepth == 0:
                     break
-            else: print("Unknown token while parsing stream: " + token)
-        else: break
+            else:
+                print("Unknown token while parsing stream: " + token)
+        else:
+            break
     generatedRegexes.clear()
 
-        
+
 def parseFile(filename):
-    #filename = filename.decode('utf8') #TODO: fix
     try:
         theFile = open(filename, 'r')
     except FileNotFoundError:
         print("Could not open " + filename + " for parsing: File not found!")
         return
-    except:
-        print("Could not open " + filename + " for parsing.")
+    except IOError:
+        print("Could not open " + filename + " for parsing: cannot open file!")
         return
-	
+
     firstChar = peek_char(theFile)
-    if (firstChar == '\xEF'):
-        bitBucket=theFile.read(3)
+    if firstChar == '\xEF':
+        theFile.read(3)  # skip 3 bytes
```

Comment:
C++ skips BOM because we read it as regular characters. Maybe python doesn't. Try with a test file please.

Reply:
Turns out that was the case: if you open a file with `encoding="utf-8-sig"` and it doesn't have BOM, it bahaves like it was open with `encoding="utf-8`, at least from what I read (test confirmed this).